Amplify/Categories/Auth/Models/AuthUserAttribute.swift

```diff
@@ -15,22 +15,54 @@ public struct AuthUserAttribute {
     }
 }
 
+/// Represents the keys used for different user attributes.
+///
 public enum AuthUserAttributeKey {
     // Attribute ref - https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html
+
+    /// Attribute key for user's address
     case address
+
+    /// Attribute key for user's birthdate
     case birthDate
+
+    /// Attribute key for user's email
     case email
+
+    /// Attribute key for user's family name
     case familyName
+
+    /// Attribute key for user's gender
     case gender
+
+    /// Attribute key for user's given name
     case givenName
+
+    /// Attribute key for user's locale
     case locale
+
+    /// Attribute key for user's middle name
     case middleName
+
+    /// Attribute key for user's name
     case name
+
+    /// Attribute key for user's nickname
     case nickname
+
+    /// Attribute key for user's phone number
     case phoneNumber
+
+    /// Attribute key for user's picture
     case picture
+
+    /// Attribute key for user's preferred user name
     case preferredUsername
+
+    /// Attribute key for providing custom attributes
```

Comment:
Note that for Cognito User Pools, the attribute either automatically prepends "custom:" or needs to have it prepended. We should specify which one, and how it's different from "unknown"